packages/common/src/items/fetch-all-pages.ts

```diff
@@ -0,0 +1,33 @@
+import { ISearchOptions, ISearchResult } from "@esri/arcgis-rest-portal";
+import { SearchableType, SearchFunction } from "../types";
+import { batch } from "../utils/batch";
+
+const MAX_NUM = 100;
+
+/**
+ * Fetches all the pages
+ * @param opts search options
+ */
+export function fetchAllPages(
+  searchFunc: SearchFunction,
+  opts: ISearchOptions,
+  pageSize = MAX_NUM
+): Promise<SearchableType[]> {
+  return searchFunc({ ...opts, num: 1 })
```

Comment:
This call is just to get the total and should be pretty quick. However, I toyed with other ideas:
- Just use the first slice to get the `total` and do things batch-wise after that. Only drawback is that the batch routine is delayed for the duration of the first request/response (which could be quite sizable for a large query).
- Allow a `limit` parameter to be passed to short-circuit this. But, that brings in the question of what should the limit be, etc.

So, I'm open to input.

Reply:
This is fine as is, but I also I like the idea of using "the first slice to get the total and do things batch-wise after that."

Reply:
Done.